common/source/docs/common-blheli32-passthru.rst

```diff
@@ -1,62 +1,149 @@
+
+This data can also be viewed in real-time using a ground station.  If using the Mission Planner go to the Flight Data screen's status tab and look for esc1_rpm.
+
+.. image:: ../../../images/dshot-realtime-esc-telem-in-mp.jpg
+    :target: ../_images/dshot-realtime-esc-telem-in-mp.jpg
+    :width: 450px
+
+.. note::
+
+   Sending BLHeli32 telemetry data to the GCS requires the telemetry connection use MAVLink2.  ArduPilot uses MAVLink2 by default on the USB port but if another port is used it may be necessary to set the SERIALx_PROTOCOL parameter to 2 (where "x" is the serial port number used for the telemetry connection).
+
+In addition, some telemetry values can be displayed on the integrated :ref:`on-board OSD <common-osd-overview>`, if your autopilot has one.
+
+.. _esc-telemetry-based-battery-monitor:
+
+Use as Battery Monitor
+======================
+
+By setting a battery monitor instance to BLHeli32 ESC  type (for example :ref:`BATT2_MONITOR<BATT2_MONITOR>` = 9), all connected BLHeli32 ESCs with connected telemetry wiring to the configured autopilot serial port, will be aggregated as a single source. The voltages reported will be averaged, the currents totaled, and the consumed current accumulated.
+
+.. _bidir-dshot:
+
+Bi-directional Dshot
+====================
+
+Newer versions of BLHeli32 (32.7) and BLHeli_S (16.73) support returning motor RPM values over the Dshot signal line. Supporting bi-directional Dshot requires exclusive use of one or more DMA channels and thus not all versions of ArduPilot support it. Versions that support bi-directional Dshot natively are `BeastH7`, `BeastF7` and `KakuteF7Mini`, other firmware versions end in "-bdshot" to indicate support for bi-directional Dshot.
+
+Setup
+-----
 
-Many brushless BLHeli ESCs offer telemetry reporting of important ESC data, such as RPM, Temperature, Current, etc. for use in OSDs or for controlling ArduPilot functions like variable center frequency noise filters. This can be via a serial connection to one of the autopilot's UART RX inputs, whose ``SERIALx_PROTOCOL`` has been set to "16" (ESC Telemetry), or directly over its control connection if it has Bi-Directional DShot capability (only available on certain BLHeli32 capable ESCs at present).
+First ensure that you have an appropriate version of BLHeli32 or BLHeli_S installed on your ESCs. The majority of ESCs do not come pre-installed with these versions. The official 32.7 version of BLHeli32 supports bi-directional Dshot. Official versions of BLHeli_S do not support bi-directional Dshot, you will need to either buy a version from `JESC <https://jflight.net/index.php?route=common/home&language=en-gb>`__ or use the unofficial version from `JazzMaverick <https://github.com/JazzMaverick/BLHeli/tree/JazzMaverick-patch-1/BLHeli_S%20SiLabs/Hex%20files%20%2016.73>`__. If you try and enable bi-directional Dshot with the wrong firmware version then unpredictable motor operation can occur.
```

Comment:
most people wont know how to check the version without instructions....the ESC should say its bi-directional....flashing the firmware is a very advanced user topic and would need a new wiki page....IF passthrough works for flashing, which I do not know...

Reply:
in Betaflight there is a requirement to use the proper Dshot rate for the notch filter's use depending on loop rates...I dont know if there is such a requirement in our notch filters( I assume we use the report from the designated bi-dir escs for mode 3)...if so, it should be mentioned here in the mode 3 notch docs

Reply:
This section has just been moved from the BLHeli telemetry page (https://ardupilot.org/plane/docs/common-dshot-blheli32-telemetry.html) with no changes because I haven't yet tested it.  Still, I've added a picture of the blheli suite with some red squares around buttons that need pressing in order to check the version.

It is possible to update the firmware using blheli pass-through but I haven't added instructions for that.

I'll leave @andyp1per to respond on the notch filter requirements.

Reply:
Firmware flashing works fine

Notch requirements should not be affected by dshot rate but DShot600 is the most tested. We cannot run at a high enough loop rate where the dshot rate matters